src/main/java/uk/gov/hmcts/reform/civil/handler/tasks/CheckUnlessOrderDeadlineEndTaskHandler.java

```diff
@@ -0,0 +1,85 @@
+package uk.gov.hmcts.reform.civil.handler.tasks;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.camunda.bpm.client.task.ExternalTask;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.civil.enums.YesOrNo;
+import uk.gov.hmcts.reform.civil.helpers.CaseDetailsConverter;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.model.genapplication.GAJudicialMakeAnOrder;
+import uk.gov.hmcts.reform.civil.service.CoreCaseDataService;
+import uk.gov.hmcts.reform.civil.service.search.CaseStateSearchService;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.time.LocalDate.now;
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.END_SCHEDULER_CHECK_UNLESS_ORDER_DEADLINE;
+import static uk.gov.hmcts.reform.civil.enums.CaseState.ORDER_MADE;
+import static uk.gov.hmcts.reform.civil.enums.dq.GeneralApplicationTypes.UNLESS_ORDER;
+
+@Slf4j
+@RequiredArgsConstructor
+@Component
+@ConditionalOnExpression("${judge.revisit.stayOrder.event.emitter.enabled:true}")
+public class CheckUnlessOrderDeadlineEndTaskHandler implements BaseExternalTaskHandler {
+
+    private final CaseStateSearchService caseSearchService;
+
+    private final CoreCaseDataService coreCaseDataService;
+
+    private final CaseDetailsConverter caseDetailsConverter;
+    private final ObjectMapper mapper;
+
+    @Override
+    public void handleTask(ExternalTask externalTask) {
+        List<CaseData> cases = getUnlessOrderCasesThatAreEndingToday();
+        log.info("Job '{}' found {} case(s)", externalTask.getTopicName(), cases.size());
+
+        cases.forEach(this::fireEventForStateChange);
+    }
+
+    private List<CaseData> getUnlessOrderCasesThatAreEndingToday() {
+        List<CaseDetails> unlessOrderCases = caseSearchService
+            .getOrderMadeGeneralApplications(ORDER_MADE, UNLESS_ORDER);
+        return unlessOrderCases.stream()
+            .map(caseDetailsConverter::toCaseData)
+            .filter(caseData -> caseData.getJudicialDecisionMakeOrder()
+                .getJudgeApproveEditOptionDateForUnlessOrder() != null
+                && caseData.getGeneralAppType().getTypes().contains(UNLESS_ORDER)
```

Comment:
We already filter for Apptype when we call cacheService at line 49, may be we can delete this condition as all application will be of type "UNLESS_ORDER"